client/blocks/login/login-form.jsx

```diff
@@ -158,6 +164,74 @@ export class LoginForm extends Component {
 		return ! socialAccountIsLinking && ! hasAccountTypeLoaded;
 	}
 
+	getSignupUrl() {
+		const {
+			oauth2Client,
+			redirectTo,
+			isGutenboarding,
+			currentRoute,
+			currentQuery,
+			pathname,
+			locale,
+		} = this.props;
+		const isOauthLogin = !! oauth2Client;
+		const langFragment = locale && locale !== 'en' ? `/${ locale }` : '';
+		const signupFlow = get( currentQuery, 'signup_flow' );
+		let signupUrl = config( 'signup_url' );
+
+		// copied from login-links.jsx
+		if (
+			// Match locales like `/log-in/jetpack/es`
+			startsWith( currentRoute, '/log-in/jetpack' )
+		) {
+			// Basic validation that we're in a valid Jetpack Authorization flow
+			if (
+				includes( get( currentQuery, 'redirect_to' ), '/jetpack/connect/authorize' ) &&
+				includes( get( currentQuery, 'redirect_to' ), '_wp_nonce' )
+			) {
+				/**
+				 * `log-in/jetpack/:locale` is reached as part of the Jetpack connection flow. In
+				 * this case, the redirect_to will handle signups as part of the flow. Use the
+				 * `redirect_to` parameter directly for signup.
+				 */
+				signupUrl = currentQuery.redirect_to;
+			} else {
+				signupUrl = '/jetpack/connect';
+			}
+		} else if ( '/jetpack-connect' === pathname ) {
+			signupUrl = '/jetpack/connect';
+		} else if ( signupFlow ) {
+			signupUrl += '/' + signupFlow;
+		}
+
+		if ( isOauthLogin && config.isEnabled( 'signup/wpcc' ) ) {
+			const oauth2Flow = isCrowdsignalOAuth2Client( oauth2Client ) ? 'crowdsignal' : 'wpcc';
+			const oauth2Params = new globalThis.URLSearchParams( {
+				oauth2_client_id: oauth2Client.id,
+				oauth2_redirect: redirectTo || '',
+			} );
+
+			signupUrl = `/start/${ oauth2Flow }?${ oauth2Params.toString() }`;
+		}
+
+		if ( isGutenboarding ) {
+			signupUrl = '/new' + langFragment;
+		}
+
+		return signupUrl;
+	}
+
+	showUsernameError() {
+		const { requestError } = this.props;
+
+		// TODO: add feature flag or Jetpack context condition
+		return (
+			requestError &&
+			requestError.field === 'usernameOrEmail' &&
+			requestError.code !== LOGIN_ERROR_UNKNOWN_USER
```

Comment:
This prevents the login form from showing an error while the signup page is loading. The transition is too fast to read anything.